### PR TITLE
Fix compile errors on Solaris

### PR DESCRIPTION
**Pull request by japinli**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@japinli Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=greenplum-db/gporca&pullRequestId=584)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=greenplum-db/gporca&pullRequestId=584) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@japinli Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=greenplum-db/gporca&pullRequestId=584)!

**Comment by asubramanya** [comment]
Thanks for your PR @japinli. GPDB is not supported on Solaris. @bradfordboyle @kmacoskey

Also note that ORCA code was merged into the GPDB repository a few days back. Any ORCA changes that should go into master and 6X_STABLE should now be made there (https://github.com/greenplum-db/gpdb). The current ORCA repository is only for PR's that should go into 5X_STABLE.

File: libgpos/include/gpos/task/IWorker.h [hunk 1]
```diff
@@ -31,7 +31,7 @@
 	while (0)
 
 
-#if (GPOS_SunOS)
+#if defined(GPOS_SunOS)
```
Comment (d) [inline libgpos/include/gpos/task/IWorker.h:34]: -1

given how this preprocessor definition is generated, it's not clear this diff is necessary. Can you elaborate in the PR message what's the error you're facing?

**Comment by zellerh** [comment]
Hi @japinli, we'll be happy to commit your PR. Here are a few questions for you: As @asubramanya pointed out, this repository is now for GPDB 5X only. Are you planning to use Orca with GPDB 5X? If so, we can commit this PR, otherwise, we would recommend that you submit a PR on GPDB master (https://github.com/greenplum-db/gpdb). You can find the relevant code in src/backend/gporca. Note that with this new model we use GNU make instead of cmake, and your build may no longer work on Solaris. In GPDB master, the src/backend/gporca/CMakeLists.txt file is only used to build test programs like gporca_test, and for use with XCode.

File: libgpos/include/gpos/task/IWorker.h (hunk 1 above)
Comment (japinli) [reply libgpos/include/gpos/task/IWorker.h:34]: @d, when i compile it on SunOS, in the `libgpos/include/gpos/config.h`, which is generated by cmake, has the following codes:

```c
// These come from CMAKE_SYSTEM_NAME
/* #undef GPOS_Linux */
/* #undef GPOS_Darwin */
#define GPOS_SunOS
```

Here, `GPOS_SunOS` is define to null, and make an error:

```
[39/1069] Building CXX object libgpos/CMakeFiles/gpos.dir/src/common/CDebugCounter.cpp.o
FAILED: libgpos/CMakeFiles/gpos.dir/src/common/CDebugCounter.cpp.o
/opt/local/bin/c++  -Dgpos_EXPORTS -I/root/gpdb-solaris/gpdb-5.25.0/include -I../libgpos/include -I../libgpopt/include -I../libnaucrates/include -Ilibgpos/include -Wall -Werror -Wextra -pedantic-errors -Wno-variadic-macros -fno-omit-frame-pointer -g -g3 -fPIC   -std=gnu++98 -MD -MT libgpos/CMakeFiles/gpos.dir/src/common/CDebugCounter.cpp.o -MF libgpos/CMakeFiles/gpos.dir/src/common/CDebugCounter.cpp.o.d -o libgpos/CMakeFiles/gpos.dir/src/common/CDebugCounter.cpp.o -c ../libgpos/src/common/CDebugCounter.cpp
In file included from /root/gpdb-solaris/gpdb-5.25.0/include/gpos/task/ITask.h:15:0,
                 from /root/gpdb-solaris/gpdb-5.25.0/include/gpos/base.h:22,
                 from /root/gpdb-solaris/gpdb-5.25.0/include/gpos/common/CDebugCounter.h:85,
                 from ../libgpos/src/common/CDebugCounter.cpp:13:
/root/gpdb-solaris/gpdb-5.25.0/include/gpos/task/IWorker.h:34:16: error: missing expression between '(' and ')'
 #if (GPOS_SunOS)
                ^
```

So, I made this change, we can also change it as follows:

```c
#ifdef GPOS_SunOS
```

**Comment by japinli** [comment]
Hi, @zellerh, I want to use gporca in GPDB 5X on SunOS.

**Comment by zellerh** [comment]
Just curious, were you able to build and run the code on Solaris after applying your fix? Given that Solaris is not a supported or tested platform, several people have expressed doubt that gpdb/gporca will work.

**Comment by japinli** [comment]
Here, I just fix the compile bug, I am going to test whether it works on Solaris.

**Comment by zellerh** [comment]
@d, I'm wondering whether you still object to the PR, given the additional info. I tried building this on my laptop and it works fine. In my case the `GPOS_SunOS` variable is not defined. Also, I don't think we use the `GPOS_CHECK_ABORT_MAX_INTERVAL_MSEC` macro defined in IWorker.h anymore, we could probably delete the whole macro definition.

File: libgpos/include/gpos/task/IWorker.h (hunk 1 above)
Comment (d) [reply libgpos/include/gpos/task/IWorker.h:34]: oooh, LGTM.

I did a quick check on our pattern of usages around the OS `#define`s, it seems `#ifdef` and `#if defined` are intermixed. So yeah I withdraw my objection

**Comment by d** [comment]
LGTM

**Pull request merged**